src/main/java/core/basesyntax/serviceimpl/DataConverterImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter<List<String>, List<FruitTransaction>> {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(line -> {
+                    String[] parts = line.split(",");
+                    return new FruitTransaction(
+                            FruitTransaction.Operation.fromCode(parts[0]),
+                            parts[1],
+                            Integer.parseInt(parts[2])
```

Comment:
It's a good practice to validate the input data. Ensure that 'parts[0]' is a valid operation code and 'parts[1]' is a non-empty string representing the fruit, and 'parts[2]' can be successfully parsed as an integer.